### PR TITLE
1inch Plugin v5 Upgrade

### PR DESCRIPTION
**Pull request by GuilaneDen**
## Smart Contracts

Smart contracts covered by this upgrade:

| Network | Version | Smart Contract |
| ---       | --- | --- |
| Ethereum  | V5  | `0x1111111254eeb25477b68fb85ed929f73a960582`|
| Polygon   | V5  | `0x1111111254eeb25477b68fb85ed929f73a960582`|

## Functions implemented:

|    Function   | Selector  | Displayed Parameters |
| ---           | ---       | --- |
|Swap (v5)| 0x12aa3caf| <table>  <tbody>  <tr> <td><code>address desc.srcToken</code></td></tr> <tr><td><code>address desc.dstToken</code></td></tr> <tr><td><code>address desc.dstReceiver</code></td></tr> <tr><td><code>uint256 desc.amount</code></td></tr> <tr><td><code>uint256 desc.minReturnAmount</code></td></tr> <tr><td><code>uint256 desc.flags</code></td></tr> </tbody> </table> |
|Unoswap (v5) | 0x0502b1c5| <table>  <tbody>  <tr><td><code>address srcToken</code></td></tr> <tr><td><code>uint256 amount</code></td></tr> <tr><td><code>uint256 minReturn</code></td></tr> </tbody> </table>|
|unoswapToWithPermit (v5) | 0x3c15fd91| <table>  <tbody> <tr><td><code>address recipient</code></td></tr> <tr><td><code>address srcToken</code></td></tr> <tr><td><code>uint256 amount</code></td></tr> <tr><td><code>uint256 minReturn</code></td></tr> </tbody> </table>|
|clipperSwap (v5)  | 0x84bd6d29| <table>  <tbody>  <tr> <td><code>address srcToken</code></td></tr> <tr><td><code>address dstToken</code></td></tr> <tr><td><code>uint256 inputAmount</code></td></tr> <tr><td><code>uint256 outputAmount</code></td></tr> </tbody> </table> |
|clipperSwapToWithPermit (v5)  | 0xc805a666| <table>  <tbody>  <tr><td><code>address recipient</code></td></tr> <tr><td><code>address srcToken</code></td></tr> <tr><td><code>address dstToken</code></td></tr> <tr><td><code>uint256 inputAmount</code></td></tr> <tr><td><code>uint256 outputAmount</code></td></tr> </tbody> </table>|
|fillOrderRFQ (v5)  | 0x3eca9c0a| <table>  <tbody>  <tr><td><code>address order.makerAsset</code></td></tr> <tr><td><code>address order.takerAsset</code></td></tr> <tr><td><code>address order.maker</code></td></tr> <tr><td><code>uint256 order.makingAmount</code></td></tr> <tr><td><code>uint256 order.takingAmount</code></td></tr> </tbody> </table>|
|fillOrderRFQToWithPermit (v5)  | 0x70ccbd31| <table>  <tbody>  <tr><td><code>address order.makerAsset</code></td></tr> <tr><td><code>address order.takerAsset</code></td></tr> <tr><td><code>address order.maker</code></td></tr> <tr><td><code>uint256 order.makingAmount</code></td></tr> <tr><td><code>uint256 order.takingAmount</code></td></tr> </tbody> </table>|

### CODE REVIEW

File: .github/workflows/ci-workflow.yml [hunk 1]
```diff
@@ -62,13 +63,14 @@ jobs:
     runs-on: ubuntu-latest
 
     container:
-      image: ghcr.io/ledgerhq/ledger-app-builder/ledger-app-scanner:latest
+      image: ghcr.io/blooo-io/ledger-app-builder/ledger-app-scanner:master
```
Comment (adrienlacombe-ledger) [inline .github/workflows/ci-workflow.yml:66]: please put back Ledger build there, ty!

**Comment by adrienlacombe-ledger** [comment]
@GuilaneDen few comments from the Donjon

- context->valid does not seem necessary to use (remove all its usage and definition)  https://github.com/LedgerHQ/app-plugin-1inch/blob/349666e3ccb9c82240c2ad85c587fcf56775e533/src/handle_init_contract.c#L19
- Code duplication should be avoided sent_network_token/received_network_token is defined twice.
- Should SWAP_V5 also be included in https://github.com/LedgerHQ/app-plugin-1inch/blob/349666e3ccb9c82240c2ad85c587fcf56775e533/src/handle_finalize.c#L22 ?

**Comment by GuilaneDen** [comment]
@adrienlacombe-ledger Thank you for the review the SWAP_V5 was indeed missing, hence a screen was not displayed. The review is implemented on the new fix.

**Comment by adrienlacombe-ledger** [comment]
thank you @GuilaneDen !

**Comment by agrojean-ledger** [comment]
@GuilaneDen @adrienlacombe-ledger I can't see a version bump in the `Makefile`, you should probably do it before I merge it.

**Comment by GuilaneDen** [comment]
@agrojean-ledger good point, I'll update the version asap.

**Comment by agrojean-ledger** [comment]
:heavy_check_mark: CI is ok.
:heavy_check_mark: Version bump.
:heavy_check_mark: Merge to develop for P4 deployment.

Ok for merge.

**Pull request merged**